typescript-selenium/TestConstants.ts

```diff
@@ -0,0 +1,37 @@
+/*********************************************************************
+ * Copyright (c) 2018 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+function readEnvAndSetValue(envProperty: any, defaultValue: string | number | boolean): any {
+    const propertyValue = envProperty
+    if (propertyValue === undefined || propertyValue === null) {
+        return defaultValue;
+    }
+
+    return propertyValue
+}
+
+
+
+export const TestConstants = {
+    TS_SELENIUM_HEADLESS: readEnvAndSetValue(process.env.TS_SELENIUM_HEADLESS, false),
+
+    TS_SELENIUM_START_WORKSPACE_TIMEOUT: readEnvAndSetValue(process.env.TS_SELENIUM_START_WORKSPACE_TIMEOUT, 240000),
+    TS_SELENIUM_LOAD_PAGE_TIMEOUT: readEnvAndSetValue(process.env.TS_SELENIUM_LOAD_PAGE_TIMEOUT, 120000),
+    TS_SELENIUM_LANGUAGE_SERVER_START_TIMEOUT: readEnvAndSetValue(process.env.TS_SELENIUM_LANGUAGE_SERVER_START_TIMEOUT, 180000),
+
+    TS_SELENIUM_DEFAULT_TIMEOUT: readEnvAndSetValue(process.env.TS_SELENIUM_DEFAULT_TIMEOUT, 20000),
+    TS_SELENIUM_DEFAULT_ATTEMPTS: readEnvAndSetValue(process.env.TS_SELENIUM_DEFAULT_ATTEMPTS, 5),
+    TS_SELENIUM_DEFAULT_POLLING: readEnvAndSetValue(process.env.TS_SELENIUM_DEFAULT_POLLING, 1000),
+
+    TS_SELENIUM_WORKSPACE_STATUS_ATTEMPTS: readEnvAndSetValue(process.env.TS_SELENIUM_WORKSPACE_STATUS_ATTEMPTS, 90),
+    TS_SELENIUM_WORKSPACE_STATUS_POLLING: readEnvAndSetValue(process.env.TS_SELENIUM_WORKSPACE_STATUS_POLLING, 10000),
+
+    TS_SELENIUM_BASE_URL: readEnvAndSetValue(process.env.TS_SELENIUM_BASE_URL, "http://che-che.192.168.99.100.nip.io")
+}
```

Comment:
`http://che-che.192.168.99.100.nip.io` still looks strange...

Reply:
it is default value which valid for minikube Che. For customization may be used the  "TS_SELENIUM_BASE_URL" environment variable

Reply:
i am not sure it that 192.168.99.100 default minikube value and nip.io suffix also can be changed it may be have many values for customizing. It can be a bit confuse our contributors. I guess something like localhost will be more clearer

Reply:
There are a lot of variants, I just put one of them

Reply:
Couldn't we make that parameter as "mandatory", as we can agree there is not right "default" value?